polkadot/runtime/common/src/xcm_sender.rs

```diff
@@ -141,6 +141,12 @@ where
 }
 
 impl<T: dmp::Config, W, P> InspectMessageQueues for ChildParachainRouter<T, W, P> {
+	fn clear_messages() {
+		let prefix = dmp::DownwardMessageQueues::<T>::final_prefix();
+		// Best effort.
+		let _ = frame_support::storage::unhashed::clear_prefix(&prefix, None, None);
```

Comment:
Why is this necessary? looks hacky :)

Reply:
Yes, there's a function [clear](https://paritytech.github.io/polkadot-sdk/master/frame_support/storage/types/struct.StorageMap.html#method.clear) on StorageMap that is the nicer API for deleting but it's meant to be used in the runtime, so you **have to** pass in a limit and then iterate until you're done. There's another function [remove_all](https://paritytech.github.io/polkadot-sdk/master/frame_support/storage/types/struct.StorageMap.html#method.remove_all) which can remove everything without a limit but it's marked as deprecated. What I ended up doing is using the implementation from `remove_all`.

What I could do is put this under a new method `remove_all_no_consensus` under StorageMap that's meant to not be used in the runtime.

Reply:
You can also put `u32::MAX` as limit?